Sources/Swift/Tools/UIRedactBuilder.swift

```diff
@@ -0,0 +1,136 @@
+#if canImport(UIKit) && !SENTRY_NO_UIKIT
+#if os(iOS) || os(tvOS)
+import Foundation
+import UIKit
+
+struct RedactRegion {
+    let rect: CGRect
+    let color: UIColor?
+    
+    init(rect: CGRect, color: UIColor? = nil) {
+        self.rect = rect
+        self.color = color
+    }
+    
+    func splitBySubtracting(region: CGRect) -> [RedactRegion] {
+        guard rect.intersects(region) else { return [self] }
+        guard !region.contains(rect) else { return [] }
+        
+        let intersectionRect = rect.intersection(region)
+        var resultRegions: [CGRect] = []
+        
+        // Calculate the top region.
+        resultRegions.append(CGRect(x: rect.minX,
+                                    y: rect.minY,
+                                    width: rect.width,
+                                    height: intersectionRect.minY - rect.minY))
+        
+        // Calculate the bottom region.
+        resultRegions.append(CGRect(x: rect.minX,
+                                    y: intersectionRect.maxY,
+                                    width: rect.width,
+                                    height: rect.maxY - intersectionRect.maxY))
+        
+        // Calculate the left region.
+        resultRegions.append(CGRect(x: rect.minX,
+                                    y: max(rect.minY, intersectionRect.minY),
+                                    width: intersectionRect.minX - rect.minX,
+                                    height: min(intersectionRect.maxY, rect.maxY) - max(rect.minY, intersectionRect.minY)))
+        
+        // Calculate the right region.
+        resultRegions.append(CGRect(x: intersectionRect.maxX,
+                                    y: max(rect.minY, intersectionRect.minY),
+                                    width: rect.maxX - intersectionRect.maxX,
+                                    height: min(intersectionRect.maxY, rect.maxY) - max(rect.minY, intersectionRect.minY)))
+        
+        return resultRegions.filter { !$0.isEmpty }.map { RedactRegion(rect: $0, color: color) }
+    }
+}
+
+class UIRedactBuilder {
+    
+    //This is a list of UIView subclasses that will be ignored during redact process
+    var ignoreClasses: [AnyClass] 
+    //This is a list of UIView subclasses that need to be redacted from screenshot
+    var redactClasses: [AnyClass]
+    
+    init() {
+        redactClasses = [ UILabel.self, UITextView.self, UITextField.self ] + [
+            "_TtCOCV7SwiftUI11DisplayList11ViewUpdater8Platform13CGDrawingView",
+            "_TtC7SwiftUIP33_A34643117F00277B93DEBAB70EC0697122_UIShapeHitTestingView",
+            "SwiftUI._UIGraphicsView", "SwiftUI.ImageLayer"
```

Comment:
`m`: Where do these class names come from? Can you add a comment to explain this?